src/placeos-driver/proxy/scheduler.cr

```diff
@@ -5,8 +5,7 @@ require "../driver_manager"
 
 class PlaceOS::Driver::Proxy::Scheduler
   class TaskWrapper
-    def initialize(@task : Tasker::Task, @schedules : Set(TaskWrapper))
-      @terminated = false
+    def initialize(@task : Tasker::Task, @callback : (TaskWrapper, Bool) -> Bool)
```

Comment:
```suggestion
    def initialize(@task : Tasker::Task, &@callback : (TaskWrapper, Bool) -> Bool)
 ```
 
 Allows `TaskWrapper` to be used like this
 
 ```crystal
 TaskWrapper.new(task) do |task_wrapper, add|
 ...
 end
 ```
 
 instead of 
 
 ```crystal
TaskWrapper.new(task, Proc(TaskWrapper, Bool, Bool).new { |wrapped, add| ... })
```

Reply:
A small suggestion would be changing `(TaskWrapper, Bool) -> Bool` to document what the bool parameter does.
Something like this + a comment

```crystal
alias Terminate = Bool

# `Terminate` arg of the callback forwards run state from the parent TaskWrapper to dependents  
def initialize(@task : Tasker::Task, &@callback : (TaskWrapper, Terminate) -> Bool)
...
```

or 

```crystal
enum State
 Terminate
 Keep
end

def initialize(@task : Tasker::Task, &@callback : (TaskWrapper, State) -> Bool)
...
```

Reply:
I've implemented the `enum` as that is much clearer
I didn't change to the `&block` as we don't need to capture a new block each time a schedule is added, saves a bit of memory